src/activity/service.test.ts

```diff
@@ -0,0 +1,64 @@
+import { LoggerMock, YagnaMock, agreementPoolServiceMock, paymentServiceMock } from "../../tests/mock";
+import { ActivityPoolService } from "./service";
+import { Activity } from "./activity";
+
+const logger = new LoggerMock();
+const yagnaApi = new YagnaMock().getApi();
+describe("Activity Pool Service", () => {
+  let activityService: ActivityPoolService;
+  beforeEach(() => {
+    activityService = new ActivityPoolService(yagnaApi, agreementPoolServiceMock, paymentServiceMock, {
+      logger,
+    });
+    logger.clear();
+  });
+
+  describe("run()", () => {
+    it("should start service", async () => {
+      await activityService.run();
+      expect(logger.logs).toContain("Activity Pool Service has started");
+      await activityService.end();
+    });
+  });
+  describe("end()", () => {
+    it("should stop service", async () => {
+      await activityService.run();
+      await activityService.end();
+      expect(logger.logs).toContain("Activity Pool Service has been stopped");
```

Comment:
Same here, testing the logger doesn't really tell us if the business logic inside works as expected. I would check here if every activity in the pool is cancelled and what happens if any of the calls to `cancel` reject?